automation/complex-prod-workflow-testV2.sh

```diff
@@ -0,0 +1,155 @@
+    then
+        echo "$1 needs its refresh token refreshed"
+        NEED_TOKEN=true
+    fi
+}
+
+launchSubmission() {
+    user=$1
+    namespace=$2
+    name=$3
+    methodConfigurationNamespace=$4
+    methodConfigurationName=$5
+    entityType=$6
+    entityName=$7
+    useCallCache=$8
+    expression=$9  #optional
+
+    echo "
+    Launching submission for:
+        user=$1
+        namespace=$2
+        name=$3
+        methodConfigurationNamespace=$4
+        methodConfigurationName=$5
+        entityType=$6
+        entityName=$7
+        useCallCache=$8
+        expression=$9
+    "
+
+     ACCESS_TOKEN=`docker run --rm -v $WORKING_DIR:/app/populate -w /app/populate broadinstitute/dsp-toolbox python get_bearer_token.py "${user}" "${JSON_CREDS}"`
+
+   # check if $9 is set for expression
+    if [ -z ${9+x} ] ; then
+        curl -X POST "https://api.firecloud.org/api/workspaces/$namespace/$name/submissions" -H "origin: https://portal.firecloud.org" -H "accept-encoding: gzip, deflate, br" -H "authorization: Bearer $ACCESS_TOKEN" -H "content-type: application/json" --data-binary "{\"methodConfigurationNamespace\":\"$methodConfigurationNamespace\",\"methodConfigurationName\":\"$methodConfigurationName\",\"entityType\":\"$entityType\",\"entityName\":\"$entityName\",\"useCallCache\":$useCallCache}" --compressed
+    else
+        curl -X POST "https://api.firecloud.org/api/workspaces/$namespace/$name/submissions" -H "origin: https://portal.firecloud.org" -H "accept-encoding: gzip, deflate, br" -H "authorization: Bearer $ACCESS_TOKEN" -H "content-type: application/json" --data-binary "{\"methodConfigurationNamespace\":\"$methodConfigurationNamespace\",\"methodConfigurationName\":\"$methodConfigurationName\",\"entityType\":\"$entityType\",\"entityName\":\"$entityName\",\"useCallCache\":$useCallCache,\"expression\":\"$expression\"}" --compressed
+    fi
+}
+
+findSubmissionID() {
+    user=$1
+    namespace=$2
+    name=$3
+
+    ACCESS_TOKEN=`docker run --rm -v $WORKING_DIR:/app/populate -w /app/populate broadinstitute/dsp-toolbox python get_bearer_token.py "${user}" "${JSON_CREDS}"`
+
+    submissionId=$(curl -X GET --header 'Accept: application/json' --header "Authorization: Bearer $ACCESS_TOKEN" "https://api.firecloud.org/api/workspaces/$namespace/$name/submissions"| jq -r '.[] | select(.status == ("Submitted")) | .submissionId')
+    #submissionId=$(curl -X GET --header 'Accept: application/json' --header "Authorization: Bearer $ACCESS_TOKEN" "https://api.firecloud.org/workspaces/$namespace/$name/submissions"| jq -r '.[] | select(.status == ("Submitted")) | .submissionId')
+}
+
+monitorSubmission() {
+    user=$1
+    namespace=$2
+    name=$3
+    submissionId=$4
+
+    ACCESS_TOKEN=`docker run --rm -v $WORKING_DIR:/app/populate -w /app/populate broadinstitute/dsp-toolbox python get_bearer_token.py "${user}" "${JSON_CREDS}"`
```

Comment:
`ACCESS_TOKEN` cmd repeated.

Reply:
Do you mean move it out of the function and to the top of the script?

Reply:
yeah i think that's a better move.